src/Components/Components/src/OwningComponentBase.cs

```diff
@@ -44,20 +44,63 @@ protected IServiceProvider ScopedServices
         }
     }
 
+    /// <summary>
+    /// Releases the service scope used by the component.
+    /// </summary>
     void IDisposable.Dispose()
+    {
+        Dispose(disposing: true);
+        GC.SuppressFinalize(this);
+    }
+
+    /// <summary>
+    /// Asynchronously releases the service scope used by the component.
+    /// </summary>
+    /// <returns>A task that represents the asynchronous dispose operation.</returns>
+    public async ValueTask DisposeAsync()
```

Comment:
Make the implementation private. Like the one for Dispose.

https://github.com/dotnet/aspnetcore/pull/62583/files#diff-82e944331c19e7b8ef290d4273e29da3e222ac5ed6df259a5d06de8dceaba6d0R50

To make an interface implementation private, you remove public and put the interface name right before the method `(in this case IAsyncDisposable.DisposeAsync)`